ts/models/conversation.ts

```diff
@@ -1185,49 +1203,96 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
-      this.set({ nickname: undefined, name: realUserName });
+      this.set({ nickname: undefined, displayNameInProfile: realUserName });
     } else {
-      this.set({ nickname: trimmed, name: realUserName });
+      this.set({ nickname: trimmed, displayNameInProfile: realUserName });
     }
 
     await this.commit();
-
-    await this.updateProfileName();
   }
-  public async setLokiProfile(newProfile: {
+
+  public async setSessionProfile(newProfile: {
     displayName?: string | null;
-    avatar?: string;
-    avatarHash?: string;
+    avatarPath?: string | null;
+    avatarImageId?: number;
   }) {
-    if (!_.isEqual(this.get('profile'), newProfile)) {
-      this.set({ profile: newProfile });
-      await this.commit();
+    let changes = false;
+
+    const existingSessionName = this.getRealSessionUsername();
+    if (newProfile.displayName !== existingSessionName && newProfile.displayName) {
+      this.set({
+        displayNameInProfile: newProfile.displayName,
+      });
+      changes = true;
     }
 
     // a user cannot remove an avatar. Only change it
-    // if you change this behavior, double check all setLokiProfile calls (especially the one in EditProfileDialog)
-    if (newProfile.avatar) {
-      await this.setProfileAvatar({ path: newProfile.avatar }, newProfile.avatarHash);
+    // if you change this behavior, double check all setSessionProfile calls (especially the one in EditProfileDialog)
+    if (newProfile.avatarPath) {
+      const originalAvatar = this.get('avatarInProfile');
+      if (!_.isEqual(originalAvatar, newProfile.avatarPath)) {
+        this.set({ avatarInProfile: newProfile.avatarPath });
+        changes = true;
+      }
+      const existingImageId = this.get('avatarImageId');
+
+      if (existingImageId !== newProfile.avatarImageId) {
+        this.set({ avatarImageId: newProfile.avatarImageId });
+        changes = true;
+      }
+    }
+
+    if (changes) {
+      await this.commit();
     }
+  }
 
-    await this.updateProfileName();
+  public setSessionDisplayNameNoCommit(newDisplayName?: string | null) {
+    const existingSessionName = this.getRealSessionUsername();
+    if (newDisplayName !== existingSessionName && newDisplayName) {
+      this.set({ displayNameInProfile: newDisplayName });
+    }
   }
-  public async updateProfileName() {
-    // Prioritise nickname over the profile display name
-    const nickname = this.getNickname();
-    const displayName = this.getLokiProfile()?.displayName;
 
-    const profileName = nickname || displayName || null;
-    await this.setProfileName(profileName);
+  /**
+   * @returns `displayNameInProfile` so the real username as defined by that user/group
+   */
+  public getRealSessionUsername(): string | undefined {
+    return this.get('displayNameInProfile');
   }
-  public getLokiProfile() {
-    return this.get('profile');
+
+  /**
+   * @returns `nickname` so the nickname we forced for that user. For a group, this returns `undefined`
```

Comment:
Suggestion: `nickname` - the nickname we forced for that user. For a group, this returns `undefined`